src/fpsolve/functional.jl

```diff
@@ -1,30 +1,52 @@
-function fpsolve!(integrator::DDEIntegrator, fpsolver::FPSolver{<:FPFunctionalConstantCache})
+function fpsolve!(fpsolver::FPSolver{<:Union{NLFunctional,NLAnderson},false},
```

Comment:
Should we overload `nlsolve!` instead of defining `fpsolve!`?

Reply:
ehh, if these are a different cache type then I think we might as well commit to `fpsolve!`